R/eml.R

```diff
@@ -112,47 +112,30 @@ pid_to_eml_physical <- function(mn, pids) {
 #'
 #' @examples
 #' \dontrun{
-#' # Generate EML physical objects for all the data in a package
-#' pkg <- get_package(mn, pid)
-#' sm <- lapply(pkg$data, function(pid) {
-#'   getSystemMetadata(mn, pid)
-#' })
+#' # Generate EML physical object from a system metadata object
+#' sm <- getSystemMetadata(mn, pid)
 #' sysmeta_to_eml_physical(sm)
 #' }
 sysmeta_to_eml_physical <- function(sysmeta) {
-  work <- function(x) {
-    phys <- new("physical")
-    phys@scope <- new("xml_attribute", "document")
+    stopifnot(is(sysmeta, "SystemMetadata"))
 
-    if (is.na(x@fileName)) {
-      phys@objectName <- new("objectName", "NA")
+    if (is.na(sysmeta@fileName)) {
+      ob_name <- "NA"
     } else {
-      phys@objectName <- new("objectName", x@fileName)
+      ob_name <- sysmeta@fileName
     }
 
-    phys@size <- new("size", format(x@size, scientific = FALSE))
-    phys@size@unit <- new("xml_attribute", "bytes")
 
-    phys@authentication <- new("ListOfauthentication", list(new("authentication", x@checksum)))
-    phys@authentication[[1]]@method <- new("xml_attribute", x@checksumAlgorithm)
+    phys <- set_physical(objectName = ob_name,
+                         size = format(sysmeta@size, scientific = FALSE),
+                         sizeUnit = "bytes",
+                         authentication = sysmeta@checksum,
+                         authMethod = sysmeta@checksumAlgorithm,
+                         url = paste0("https://cn.dataone.org/cn/v2/resolve/", sysmeta@identifier))
 
-    phys@dataFormat <- new("dataFormat")
-    phys@dataFormat@externallyDefinedFormat <- new("externallyDefinedFormat")
-    phys@dataFormat@externallyDefinedFormat@formatName <- x@formatId
-
-    phys@distribution <- new("ListOfdistribution", list(new("distribution")))
```

Comment:
Did setting `distribution` get dropped from the code here?

Reply:
no, it is set using the `url` argument in `set_physical`